client/my-sites/hosting/sftp-card/index.js

```diff
@@ -158,15 +124,21 @@ const SFTPCard = ( { translate, siteId } ) => {
 					</tbody>
 				</table>
 			) }
-			{ loading && ! noSftpUser && <Spinner /> }
+			{ loading && <Spinner /> }
 		</Card>
 	);
 };
 
 export default connect( state => {
 	const siteId = getSelectedSiteId( state );
+	const sftpDetails = requestAtomicSFTPDetails( siteId );
```

Comment:
I guess the cache isn't working, this is firing way too much, probably on each render. Freshness was set to ` 5 * 60 * 1000,`

Reply:
Oh since this is a 404, I think this never gets set into cache as lastUpdated, since it counts as an "error".

Reply:
if this is causing non-idempotent changes on the backend (such as `POST`ing to a remote service) then we really can't run it from within `mapStateToProps` as there is no guarantee that this won't run endlessly. I see that we have a 5-minute freshness set but if we're creating users on the remote site we don't want to endlessly create them just because someone has a Calypso window open

maybe we cleaned up the backend? is the `GET` safe and idempotent again? (meaning we can call it 1000 times in a row without causing any harm or different results) if so this is fine. if not we would want to gate it behind a user input such as clicking on a button

Reply:
Summarizing some slack chats @dmsnell found that this request is already returning 200s but we're throwing an error during the request parsing, due to the extra layers of request envelopes! 

Let's tidy up the parsing and file a new issue to fix this for http-data that documents this behavior, since this is unexpected that it'd fire this many times in this error case.

Reply:
Oh hmm maybe that's not it I don't think we explicitly return a different status code yet in https://code.a8c.com/D34243

We have a different envelope mechanism. If we pass the query param `_envelope` this calls `envelope_response` from https://developer.wordpress.org/reference/classes/wp_rest_server/envelope_response/

Which I believe helps with our wpcom auth iframe handling.

Reply:
D34660-code should fix this.